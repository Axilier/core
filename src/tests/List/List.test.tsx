import React from 'react';
import { mount } from 'enzyme';
import { List, ListGroup, ListItem } from '../../index';
import { expectToThrow } from '../expectToThrow';

const list = mount(
    <List>
        <ListItem key={'test-1'}>test</ListItem>
        <ListGroup
            key={'test-2'}
            listItem={
                <ListItem>
                    <></>
                </ListItem>
            }
        >
            <ListItem key={'test-2-1'}>test1</ListItem>
            <ListItem key={'test-2-2'}>test2</ListItem>
        </ListGroup>
    </List>,
);
const cleanUp = () => list.setProps({});

describe('when I render the List in an application', () => {
    it('should render correctly', () => {
        expect(list).toMatchSnapshot();
    });
    describe('when I render a non ListItem component as a child', () => {
        it('should cause an Error that it must only have ListItems as children', () => {
            expectToThrow(
                () =>
                    mount(
                        <List>
                            <div />
                        </List>,
                    ),
                'Children must be of type ListItem',
            );
        });
    });
    describe('when `width` is not set', () => {
        it('should have a width of 100%', () => {
            expect(list.childAt(0).props().style).toHaveProperty(
                'width',
                '100%',
            );
        });
    });
    describe('when the `width` is set', () => {
        beforeAll(() => list.setProps({ width: '200px' }));
        it('should use that width as the lists width', () => {
            expect(list.childAt(0).props().style).toHaveProperty(
                'width',
                '200px',
            );
        });
        afterAll(cleanUp);
    });
});
