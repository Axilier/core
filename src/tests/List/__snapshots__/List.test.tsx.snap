// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when I render the List in an application should render correctly 1`] = `
<List>
  <Layout
    orientation="column"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "width": "100%",
        }
      }
    >
      <ListItem
        key="test-1"
      >
        <Layout
          className="listItem"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          orientation="row"
          style={
            Object {
              "backgroundColor": "#F9F9F9",
              "paddingLeft": "30px",
            }
          }
        >
          <div
            className="listItem"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "#F9F9F9",
                "display": "flex",
                "flexDirection": "row",
                "paddingLeft": "30px",
              }
            }
          >
            <div
              className="listItemChildren"
            >
              test
            </div>
            <div
              className="endFragment"
            >
              <div
                style={
                  Object {
                    "height": "16px",
                    "width": "16px",
                  }
                }
              />
            </div>
          </div>
        </Layout>
      </ListItem>
      <ListGroup
        key="test-2"
        listItem={
          <ListItem>
            <React.Fragment />
          </ListItem>
        }
      >
        <ListItem
          dropdown={false}
          onClick={[Function]}
        >
          <Layout
            className="listItem"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            orientation="row"
            style={
              Object {
                "backgroundColor": "#F9F9F9",
                "paddingLeft": "30px",
              }
            }
          >
            <div
              className="listItem"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "#F9F9F9",
                  "display": "flex",
                  "flexDirection": "row",
                  "paddingLeft": "30px",
                }
              }
            >
              <div
                className="listItemChildren"
              />
              <div
                className="endFragment"
              >
                <svg
                  className="dropdown-test"
                  fill="black"
                  height="16"
                  style={
                    Object {
                      "transform": "none",
                    }
                  }
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.204 5h9.592L8 10.481 3.204 5zm-.753.659 4.796 5.48a1 1 0 0 0 1.506 0l4.796-5.48c.566-.647.106-1.659-.753-1.659H3.204a1 1 0 0 0-.753 1.659z"
                  />
                </svg>
              </div>
            </div>
          </Layout>
        </ListItem>
        <div
          className="listGroup"
          style={
            Object {
              "height": "0px",
            }
          }
        >
          <List>
            <Layout
              orientation="column"
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  }
                }
              >
                <ListItem
                  child={true}
                  key="test-2-1"
                >
                  <Layout
                    className="listItem"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    orientation="row"
                    style={
                      Object {
                        "backgroundColor": "#F9F9F9",
                        "paddingLeft": "40px",
                      }
                    }
                  >
                    <div
                      className="listItem"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#F9F9F9",
                          "display": "flex",
                          "flexDirection": "row",
                          "paddingLeft": "40px",
                        }
                      }
                    >
                      <div
                        className="listItemChildren"
                      >
                        test1
                      </div>
                      <div
                        className="endFragment"
                      >
                        <div
                          style={
                            Object {
                              "height": "16px",
                              "width": "16px",
                            }
                          }
                        />
                      </div>
                    </div>
                  </Layout>
                </ListItem>
                <ListItem
                  child={true}
                  key="test-2-2"
                >
                  <Layout
                    className="listItem"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    orientation="row"
                    style={
                      Object {
                        "backgroundColor": "#F9F9F9",
                        "paddingLeft": "40px",
                      }
                    }
                  >
                    <div
                      className="listItem"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#F9F9F9",
                          "display": "flex",
                          "flexDirection": "row",
                          "paddingLeft": "40px",
                        }
                      }
                    >
                      <div
                        className="listItemChildren"
                      >
                        test2
                      </div>
                      <div
                        className="endFragment"
                      >
                        <div
                          style={
                            Object {
                              "height": "16px",
                              "width": "16px",
                            }
                          }
                        />
                      </div>
                    </div>
                  </Layout>
                </ListItem>
              </div>
            </Layout>
          </List>
        </div>
      </ListGroup>
    </div>
  </Layout>
</List>
`;
